CMakeLists.txt

```
@@ -656,6 +656,8 @@ if(WITH_GRAFANA)
   add_subdirectory(monitoring/grafana/dashboards)
 endif()
 
+option(WITH_BOOST_VALGRIND "Boost support for valgrind" OFF)
```

```suggestion
CMAKE_DEPENDENT_OPTION(WITH_BOOST_VALGRIND "Boost support for valgrind" OFF
  "NOT WITH_SYSTEM_BOOST" OFF)
```

as IIUC, this option is valid only when boost is compiled with valgrind=on.

---

Done!